src/Microsoft.Framework.PropertyHelper.Internal/PropertyHelper.cs

```diff
@@ -0,0 +1,225 @@
+// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+
+namespace Microsoft.Framework.Internal
+{
+    public class PropertyHelper
+    {
+        // Delegate type for a by-ref property getter
+        private delegate TValue ByRefFunc<TDeclaringType, TValue>(ref TDeclaringType arg);
+
+        private static readonly MethodInfo CallPropertyGetterOpenGenericMethod =
+            typeof(PropertyHelper).GetTypeInfo().GetDeclaredMethod("CallPropertyGetter");
+
+        private static readonly MethodInfo CallPropertyGetterByReferenceOpenGenericMethod =
+            typeof(PropertyHelper).GetTypeInfo().GetDeclaredMethod("CallPropertyGetterByReference");
+
+        private static readonly MethodInfo CallPropertySetterOpenGenericMethod =
+            typeof(PropertyHelper).GetTypeInfo().GetDeclaredMethod("CallPropertySetter");
+
+        private static readonly ConcurrentDictionary<Type, PropertyHelper[]> ReflectionCache =
+            new ConcurrentDictionary<Type, PropertyHelper[]>();
+
+        private readonly Func<object, object> _valueGetter;
+
+        /// <summary>
+        /// Initializes a fast property helper.
+        ///
+        /// This constructor does not cache the helper. For caching, use GetProperties.
```

Comment:
nit: blank lines in doc comments are meaningless.  might need `<para>...</para>` around these two sections.